src/util/log.d

```diff
@@ -298,49 +304,47 @@ abstract class Logger
     }
 }
 
-class FileLogger(alias Layout) : Logger
+class FileLogger(Layout) : Logger
 {
-    // must be static to be thread-local
-    private static Appender!(char[]) buffer;
-
     private File file;
 
     // store cause it's awkward to derive in fileDescriptor, which must be signal-safe
     private int fileno;
 
-    this(string name, uint levels = LogLevel.info.orAbove)
+    private Layout layout;
+
+    mixin CachedAppender;
+
+    this(Layout layout, string name, uint levels = LogLevel.info.orAbove)
+    in (layout !is null)
     {
         super(levels);
-        file = File(name, "ab");
-        fileno = file.fileno;
+        this.file = File(name, "ab");
+        this.fileno = file.fileno;
+        this.layout = layout;
     }
 
-    this(File file, uint levels = LogLevel.info.orAbove)
+    this(Layout layout, File file, uint levels = LogLevel.info.orAbove)
+    in (layout !is null)
     {
         super(levels);
         this.file = file;
-        fileno = this.file.fileno;
+        this.fileno = this.file.fileno;
+        this.layout = layout;
     }
 
     override void append(const ref EventInfo eventInfo,
         scope void delegate(scope Sink sink) putMessage)
     {
-        import std.algorithm.mutation : swap;
+        auto appender = getAppender;
 
-        // avoid problems if toString functions call log - "borrow" static buffer
-        Appender!(char[]) buffer;
-
-        buffer.swap(this.buffer);
-
-        // put it back on exit, so the next call can use it
         scope(exit)
         {
-            buffer.clear;
-            buffer.swap(this.buffer);
+            clearAppender;
```

Comment:
Especially because `clear...` calls `clear` conditionally:
find a better pair of terms for "borrow" and "put back".

Reply:
removed CachedAppender completely